wp-job-manager-functions.php

```diff
@@ -303,66 +303,173 @@ function _wpjm_shuffle_featured_post_results_helper( $a, $b ) {
-			} else {
-				// No meta keys defined, search all post meta value.
-				$conditions[] = "{$wpdb->posts}.ID IN ( SELECT post_id FROM {$wpdb->postmeta} WHERE meta_value LIKE '%" . esc_sql( $job_manager_keyword ) . "%' )";
+				return $conditions;
 			}
-		}
 
-		// Search taxonomy.
-		$conditions[] = "{$wpdb->posts}.ID IN ( SELECT object_id FROM {$wpdb->term_relationships} AS tr LEFT JOIN {$wpdb->term_taxonomy} AS tt ON tr.term_taxonomy_id = tt.term_taxonomy_id LEFT JOIN {$wpdb->terms} AS t ON tt.term_id = t.term_id WHERE t.name LIKE '%" . esc_sql( $job_manager_keyword ) . "%' )";
+			/**
+			 * Constructs SQL clauses that return posts which include or exclude the search term in the provided columns.
+			 *
+			 * @see WP_Query::parse_search()
+			 *
+			 * @param string $search_term     The search term to match.
+			 * @param bool   $is_excluding    Whether posts that match the search term should be excluded.
+			 * @param string $wildcard_search The wildcard character or empty string for exact matches.
+			 * @param array  $search_columns   The columns to check.
+			 *
+			 * @return array The SQL clauses.
+			 */
+			function job_manager_construct_post_conditions( $search_term, $is_excluding, $wildcard_search, $search_columns ) {
+				global $wpdb;
+
+				if ( $is_excluding ) {
+					$like_op = 'NOT LIKE';
+				} else {
+					$like_op = 'LIKE';
+				}
+
+				$like = $wildcard_search . $wpdb->esc_like( $search_term ) . $wildcard_search;
+
+				$conditions = [];
+				foreach ( $search_columns as $search_column ) {
+					//phpcs:disabled WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Variables are safe or escaped.
+					$conditions[] = $wpdb->prepare( "( {$wpdb->posts}.$search_column $like_op %s )", $like );
+				}
+
+				$allow_query_attachment_by_filename = apply_filters( 'wp_allow_query_attachment_by_filename', false );
+				if ( ! empty( $allow_query_attachment_by_filename ) ) {
+					$conditions[] = $wpdb->prepare( "(sq1.meta_value $like_op %s)", $like );
+					//phpcs:enabled WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				}
+
+				return $conditions;
+			}
+		}
 
 		/**
-		 * Filters the conditions to use when querying job listings. Resulting array is joined with OR statements.
-		 *
-		 * @since 1.26.0
-		 *
-		 * @param array  $conditions          Conditions to join by OR when querying job listings.
-		 * @param string $job_manager_keyword Search query.
+		 * This function aims to provide similar search functionality with WP core while also including meta and taxonomy terms
+		 * in the searched columns. The functionality of WP_Query::parse_search is replicated but with additional SQL
+		 * clauses which are generated in the job_manager_construct_secondary_conditions function.
 		 */
-		$conditions = apply_filters( 'job_listing_search_conditions', $conditions, $job_manager_keyword );
-		if ( empty( $conditions ) ) {
-			return $search;
+		$default_search_columns = [ 'post_title', 'post_excerpt', 'post_content' ];
```

Comment:
One more big improvement to the search would be to prioritize matches in the title — and generally to sort results by relevance. Maybe we could do multiple searches, first just in the post title, then in content, metas? It could actually be faster for the first pages in some cases, since if there are enough results in the title, we don't need to do the content/meta searches. 

Just bringing it up to discuss here, it'd be its own PR probably as there would be some extra complexity, like handling pagination. Could also do some [weighting](https://stackoverflow.com/questions/6496866/best-way-to-do-a-weighted-search-over-multiple-fields-in-mysql) maybe?